{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess\n",
    "from pathlib import Path\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <23825659> is submitted to queue <short>.\n",
      "Job <23825660> is submitted to queue <short>.\n",
      "Job <23825661> is submitted to queue <short>.\n",
      "Job <23825662> is submitted to queue <short>.\n",
      "Job <23825663> is submitted to queue <short>.\n",
      "Job <23825664> is submitted to queue <short>.\n",
      "Job <23825665> is submitted to queue <short>.\n",
      "Job <23825666> is submitted to queue <short>.\n",
      "Job <23825667> is submitted to queue <short>.\n",
      "Job <23825668> is submitted to queue <short>.\n",
      "Job <23825669> is submitted to queue <short>.\n",
      "Job <23825670> is submitted to queue <short>.\n",
      "Job <23825671> is submitted to queue <short>.\n",
      "Job <23825672> is submitted to queue <short>.\n",
      "Job <23825673> is submitted to queue <short>.\n",
      "Job <23825674> is submitted to queue <short>.\n",
      "Job <23825675> is submitted to queue <short>.\n",
      "Job <23825676> is submitted to queue <short>.\n",
      "Job <23825677> is submitted to queue <short>.\n",
      "Job <23825678> is submitted to queue <short>.\n",
      "Job <23825679> is submitted to queue <short>.\n",
      "Job <23825680> is submitted to queue <short>.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <23825681> is submitted to queue <short>.\n",
      "Job <23825682> is submitted to queue <short>.\n",
      "Job <23825683> is submitted to queue <short>.\n",
      "Job <23825684> is submitted to queue <short>.\n",
      "Job <23825685> is submitted to queue <short>.\n",
      "Job <23825686> is submitted to queue <short>.\n",
      "Job <23825687> is submitted to queue <short>.\n",
      "Job <23825688> is submitted to queue <short>.\n",
      "Job <23825689> is submitted to queue <short>.\n",
      "Job <23825690> is submitted to queue <short>.\n",
      "Job <23825691> is submitted to queue <short>.\n",
      "Job <23825692> is submitted to queue <short>.\n",
      "Job <23825693> is submitted to queue <short>.\n",
      "Job <23825694> is submitted to queue <short>.\n",
      "Job <23825695> is submitted to queue <short>.\n",
      "Job <23825696> is submitted to queue <short>.\n",
      "Job <23825697> is submitted to queue <short>.\n",
      "Job <23825698> is submitted to queue <short>.\n",
      "Job <23825699> is submitted to queue <short>.\n",
      "Job <23825700> is submitted to queue <short>.\n",
      "Job <23825701> is submitted to queue <short>.\n",
      "Job <23825702> is submitted to queue <short>.\n",
      "Job <23825703> is submitted to queue <short>.\n",
      "Job <23825704> is submitted to queue <short>.\n",
      "Job <23825705> is submitted to queue <short>.\n",
      "Job <23825706> is submitted to queue <short>.\n",
      "Job <23825707> is submitted to queue <short>.\n",
      "Job <23825708> is submitted to queue <short>.\n",
      "Job <23825709> is submitted to queue <short>.\n",
      "Job <23825710> is submitted to queue <short>.\n",
      "Job <23825711> is submitted to queue <short>.\n",
      "Job <23825712> is submitted to queue <short>.\n",
      "Job <23825713> is submitted to queue <short>.\n",
      "Job <23825714> is submitted to queue <short>.\n",
      "Job <23825715> is submitted to queue <short>.\n"
     ]
    }
   ],
   "source": [
    "name = \"htseq_STAR\"  \n",
    "out_dir = Path(f\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/{name}\")\n",
    "out_dir.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "log_dir = Path(\n",
    "    f\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/log/{name}\")\n",
    "log_dir.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "path_lsf = Path(\n",
    "    f\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/lsf/{name}\")\n",
    "path_lsf.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "\n",
    "for file in Path(\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/data/alignments_clean_STAR\").glob(\"*Aligned.sortedByCoord.out.bam\"):\n",
    "    filename = str(file.name).split(\"Aligned.sortedByCoord.out.bam\")[0]\n",
    "    file_sorted = file.parent.joinpath(filename+'_PicardSorted.bam')\n",
    "    if filename.startswith(\"2\") or filename.startswith(\"3\"):\n",
    "        command = f\"#BSUB -W 6:00\\n\" \\\n",
    "                    f\"#BSUB -q medium\\n\" \\\n",
    "                    f\"#BSUB -n 12\\n\" \\\n",
    "                    f\"#BSUB -M 128\\n\" \\\n",
    "                    f\"#BSUB -R rusage[mem=128]\\n\" \\\n",
    "                    \"#BSUB –u nkdang@mdanderson.org\\n\" \\\n",
    "                    f\"#BSUB -J htseq_{filename}\\n\" \\\n",
    "                    f\"#BSUB -o {log_dir.joinpath(filename+'.log')}\\n\" \\\n",
    "                    f\"#BSUB –cwd /rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects\\n\" \\\n",
    "                    f\"picard SortSam INPUT={file} OUTPUT={file_sorted} SORT_ORDER=coordinate\\n\" \\\n",
    "                    f\"picard BuildBamIndex INPUT={file_sorted}\\n\" \\\n",
    "                    f\"htseq-count -f bam -s reverse -r pos {file_sorted} ref/gencode.vM31.primary_assembly.annotation.gtf > {out_dir.joinpath(filename+'.count')}\"\n",
    "    else:\n",
    "        command = f\"#BSUB -W 6:00\\n\" \\\n",
    "                    f\"#BSUB -q medium\\n\" \\\n",
    "                    f\"#BSUB -n 12\\n\" \\\n",
    "                    f\"#BSUB -M 128\\n\" \\\n",
    "                    f\"#BSUB -R rusage[mem=128]\\n\" \\\n",
    "                    \"#BSUB –u nkdang@mdanderson.org\\n\" \\\n",
    "                    f\"#BSUB -J htseq_{filename}\\n\" \\\n",
    "                    f\"#BSUB -o {log_dir.joinpath(filename+'.log')}\\n\" \\\n",
    "                    f\"#BSUB –cwd /rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects\\n\" \\\n",
    "                    f\"picard SortSam INPUT={file} OUTPUT={file_sorted} SORT_ORDER=coordinate\\n\" \\\n",
    "                    f\"picard BuildBamIndex INPUT={file_sorted}\\n\" \\\n",
    "                    f\"htseq-count -f bam -s reverse -r pos {file_sorted} ref/gencode.v42.primary_assembly.annotation.gtf > {out_dir.joinpath(filename+'.count')}\"\n",
    "    with open(path_lsf.joinpath(f'{filename}.lsf'), \"w+\") as out:\n",
    "        out.writelines(command)\n",
    "\n",
    "    command = f\"bsub < {path_lsf.joinpath(f'{filename}.lsf')}\"\n",
    "    subprocess.Popen(command, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <23838410> is submitted to queue <medium>.\n",
      "Job <23838411> is submitted to queue <medium>.\n",
      "Job <23838412> is submitted to queue <medium>.\n",
      "Job <23838413> is submitted to queue <medium>.\n",
      "Job <23838414> is submitted to queue <medium>.\n",
      "Job <23838415> is submitted to queue <medium>.\n",
      "Job <23838416> is submitted to queue <medium>.\n",
      "Job <23838417> is submitted to queue <medium>.\n",
      "Job <23838418> is submitted to queue <medium>.\n",
      "Job <23838419> is submitted to queue <medium>.\n",
      "Job <23838420> is submitted to queue <medium>.\n",
      "Job <23838421> is submitted to queue <medium>.\n",
      "Job <23838422> is submitted to queue <medium>.\n",
      "Job <23838423> is submitted to queue <medium>.\n",
      "Job <23838424> is submitted to queue <medium>.\n",
      "Job <23838425> is submitted to queue <medium>.\n",
      "Job <23838426> is submitted to queue <medium>.\n",
      "Job <23838427> is submitted to queue <medium>.\n",
      "Job <23838428> is submitted to queue <medium>.\n",
      "Job <23838429> is submitted to queue <medium>.\n",
      "Job <23838430> is submitted to queue <medium>.\n",
      "Job <23838431> is submitted to queue <medium>.\n",
      "Job <23838432> is submitted to queue <medium>.\n",
      "Job <23838433> is submitted to queue <medium>.\n",
      "Job <23838434> is submitted to queue <medium>.\n",
      "Job <23838435> is submitted to queue <medium>.\n",
      "Job <23838436> is submitted to queue <medium>.\n",
      "Job <23838437> is submitted to queue <medium>.\n",
      "Job <23838438> is submitted to queue <medium>.\n",
      "Job <23838439> is submitted to queue <medium>.\n",
      "Job <23838440> is submitted to queue <medium>.\n",
      "Job <23838441> is submitted to queue <medium>.\n",
      "Job <23838442> is submitted to queue <medium>.\n",
      "Job <23838443> is submitted to queue <medium>.\n",
      "Job <23838444> is submitted to queue <medium>.\n",
      "Job <23838445> is submitted to queue <medium>.\n",
      "Job <23838446> is submitted to queue <medium>.\n",
      "Job <23838447> is submitted to queue <medium>.\n",
      "Job <23838449> is submitted to queue <medium>.\n",
      "Job <23838450> is submitted to queue <medium>.\n",
      "Job <23838451> is submitted to queue <medium>.\n",
      "Job <23838452> is submitted to queue <medium>.\n",
      "Job <23838453> is submitted to queue <medium>.\n",
      "Job <23838454> is submitted to queue <medium>.\n",
      "Job <23838455> is submitted to queue <medium>.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <23838456> is submitted to queue <medium>.\n",
      "Job <23838457> is submitted to queue <medium>.\n",
      "Job <23838458> is submitted to queue <medium>.\n",
      "Job <23838459> is submitted to queue <medium>.\n",
      "Job <23838460> is submitted to queue <medium>.\n",
      "Job <23838461> is submitted to queue <medium>.\n",
      "Job <23838462> is submitted to queue <medium>.\n",
      "Job <23838463> is submitted to queue <medium>.\n",
      "Job <23838464> is submitted to queue <medium>.\n",
      "Job <23838465> is submitted to queue <medium>.\n",
      "Job <23838466> is submitted to queue <medium>.\n",
      "Job <23838467> is submitted to queue <medium>.\n"
     ]
    }
   ],
   "source": [
    "name = \"htseq_all\"  \n",
    "out_dir = Path(f\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/{name}\")\n",
    "out_dir.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "log_dir = Path(\n",
    "    f\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/log/{name}\")\n",
    "log_dir.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "path_lsf = Path(\n",
    "    f\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/lsf/{name}\")\n",
    "path_lsf.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "\n",
    "for file in Path(\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/data/alignments_clean_TOPHAT\").glob(\"*/accepted_hits.bam\"):\n",
    "    filename = file.parent.name   \n",
    "    file_sorted = file.parent.joinpath(filename+'_SamSorted.bam')\n",
    "    if filename.startswith(\"2\") or filename.startswith(\"3\"):\n",
    "        command = f\"#BSUB -W 6:00\\n\" \\\n",
    "                    f\"#BSUB -q medium\\n\" \\\n",
    "                    f\"#BSUB -n 4\\n\" \\\n",
    "                    f\"#BSUB -M 128\\n\" \\\n",
    "                    f\"#BSUB -R rusage[mem=128]\\n\" \\\n",
    "                    \"#BSUB –u nkdang@mdanderson.org\\n\" \\\n",
    "                    f\"#BSUB -J htseq_{filename}\\n\" \\\n",
    "                    f\"#BSUB -o {log_dir.joinpath(filename+'.log')}\\n\" \\\n",
    "                    f\"#BSUB –cwd /rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects\\n\" \\\n",
    "                    f\"htseq-count --nonunique all --max-reads-in-buffer 60000000 -s reverse -r pos {file_sorted} ref/gencode.vM31.primary_assembly.annotation.gtf > {out_dir.joinpath(filename+'.count')}\"\n",
    "                    # f\"featureCounts -M --fraction -s 2 -T 4 -p -t exon -g gene_id -a ref/gencode.vM31.primary_assembly.annotation.gtf -o {out_dir.joinpath(filename+'.count')} {file}\"\n",
    "                    # f\"samtools sort -m 16G -o {file_sorted} -O bam -n -@ 4 {file}\\n\" \\\n",
    "                    # f\"samtools index {file_sorted}\\n\" \\\n",
    "    else:\n",
    "        command = f\"#BSUB -W 6:00\\n\" \\\n",
    "                    f\"#BSUB -q medium\\n\" \\\n",
    "                    f\"#BSUB -n 4\\n\" \\\n",
    "                    f\"#BSUB -M 128\\n\" \\\n",
    "                    f\"#BSUB -R rusage[mem=128]\\n\" \\\n",
    "                    \"#BSUB –u nkdang@mdanderson.org\\n\" \\\n",
    "                    f\"#BSUB -J htseq_{filename}\\n\" \\\n",
    "                    f\"#BSUB -o {log_dir.joinpath(filename+'.log')}\\n\" \\\n",
    "                    f\"#BSUB –cwd /rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects\\n\" \\\n",
    "                    f\"htseq-count --nonunique all --max-reads-in-buffer 60000000 -s reverse -r pos {file_sorted} ref/gencode.v42.primary_assembly.annotation.gtf > {out_dir.joinpath(filename+'.count')}\"\n",
    "                    # f\"featureCounts -M --fraction -s 2 -T 4 -p -t exon -g gene_id -a ref/gencode.v42.primary_assembly.annotation.gtf -o {out_dir.joinpath(filename+'.count')} {file}\"\n",
    "                    # f\"samtools sort -m 16G -o {file_sorted} -O bam -n -@ 4 {file}\\n\" \\\n",
    "                    # f\"samtools index {file_sorted}\\n\" \\\n",
    "\n",
    "    with open(path_lsf.joinpath(f'{filename}.lsf'), \"w+\") as out:\n",
    "        out.writelines(command)\n",
    "\n",
    "    command = f\"bsub < {path_lsf.joinpath(f'{filename}.lsf')}\"\n",
    "    subprocess.Popen(command, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Popen: returncode: None args: 'bsub < /rsrch4/home/mol_cgenesis/EMC_BIC_rsr...>"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <23852335> is submitted to queue <medium>.\n",
      "Job <23852336> is submitted to queue <medium>.\n"
     ]
    }
   ],
   "source": [
    "name = \"featureCounts\"  \n",
    "out_dir = Path(f\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/{name}\")\n",
    "out_dir.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "log_dir = Path(\n",
    "    f\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/log/{name}\")\n",
    "log_dir.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "path_lsf = Path(\n",
    "    f\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/lsf/{name}\")\n",
    "path_lsf.mkdir(parents=True, exist_ok=True)\n",
    "\n",
    "file_mouse = \"\"\n",
    "file_human = \"\"\n",
    "for file in Path(\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/data/alignments_clean_TOPHAT\").glob(\"*/accepted_hits.bam\"):\n",
    "    filename = file.parent.name   \n",
    "    if filename.startswith(\"2\") or filename.startswith(\"3\"):\n",
    "        file_mouse += \" \" + str(file)\n",
    "    else:\n",
    "        file_human += \" \" + str(file)\n",
    "command = f\"#BSUB -W 24:00\\n\" \\\n",
    "                    f\"#BSUB -q medium\\n\" \\\n",
    "                    f\"#BSUB -n 12\\n\" \\\n",
    "                    f\"#BSUB -M 128\\n\" \\\n",
    "                    f\"#BSUB -R rusage[mem=128]\\n\" \\\n",
    "                    \"#BSUB –u nkdang@mdanderson.org\\n\" \\\n",
    "                    f\"#BSUB -J featureCounts_human\\n\" \\\n",
    "                    f\"#BSUB -o {log_dir.joinpath('featureCounts_human.log')}\\n\" \\\n",
    "                    f\"#BSUB –cwd /rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects\\n\" \\\n",
    "                    f\"featureCounts -M --fraction -s 2 -T 4 -p -t exon -g gene_id -a ref/gencode.v42.primary_assembly.annotation.gtf -o {out_dir.joinpath('humans.csv')}{file_human}\"\n",
    "with open(path_lsf.joinpath(f'humans.lsf'), \"w+\") as out:\n",
    "    out.writelines(command)\n",
    "\n",
    "command = f\"bsub < {path_lsf.joinpath(f'humans.lsf')}\"\n",
    "subprocess.Popen(command, shell=True)\n",
    "\n",
    "command = f\"#BSUB -W 24:00\\n\" \\\n",
    "            f\"#BSUB -q medium\\n\" \\\n",
    "            f\"#BSUB -n 12\\n\" \\\n",
    "            f\"#BSUB -M 128\\n\" \\\n",
    "            f\"#BSUB -R rusage[mem=128]\\n\" \\\n",
    "            \"#BSUB –u nkdang@mdanderson.org\\n\" \\\n",
    "            f\"#BSUB -J featureCounts_mouse\\n\" \\\n",
    "            f\"#BSUB -o {log_dir.joinpath('featureCounts_mouse.log')}\\n\" \\\n",
    "            f\"#BSUB –cwd /rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects\\n\" \\\n",
    "            f\"featureCounts -M --fraction -s 2 -T 4 -p -t exon -g gene_id -a ref/gencode.vM31.primary_assembly.annotation.gtf -o {out_dir.joinpath('mouses.csv')}{file_mouse}\"\n",
    "with open(path_lsf.joinpath(f'mouses.lsf'), \"w+\") as out:\n",
    "    out.writelines(command)\n",
    "\n",
    "command = f\"bsub < {path_lsf.joinpath(f'mouses.lsf')}\"\n",
    "subprocess.Popen(command, shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "results = pd.DataFrame()\n",
    "for file in Path(\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/htseq\").glob(\"*\"):\n",
    "    # print(file)\n",
    "    if file.name==\"humans.count\" or file.name==\"mouses.count\":\n",
    "        continue\n",
    "    if not (file.name.startswith(\"2\") or file.name.startswith(\"3\")):\n",
    "        df = pd.read_csv(file, sep='\\t', header=None, names=['gene',file.name.split(\".\")[0]]).iloc[:-5,:]\n",
    "        if len(results)==0:\n",
    "            results = df\n",
    "        else:\n",
    "            results = results.merge(df, on='gene', how='outer')\n",
    "results = results.fillna(0).set_index('gene')\n",
    "results = results.astype(int)\n",
    "results = results.reindex(sorted(results.columns), axis=1)\n",
    "results.to_csv(\"humans.count\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16:13:46.909 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/rsrch4/home/mol_cgenesis/nkdang/mambaforge/envs/rnaseq/share/picard-2.18.7-2/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Feb 21 16:13:46 CST 2023] SortSam INPUT=new/data/alignments_clean_TOPHAT/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1/accepted_hits.bam OUTPUT=new/data/alignments_clean_TOPHAT/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1_sort.bam SORT_ORDER=coordinate    VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Tue Feb 21 16:13:46 CST 2023] Executing as nkdang@ldragon1 on Linux 3.10.0-1160.15.2.el7.x86_64 amd64; OpenJDK 64-Bit Server VM 1.8.0_332-b09; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.7-SNAPSHOT\n",
      "INFO\t2023-02-21 16:15:20\tSortSam\tRead    10,000,000 records.  Elapsed time: 00:01:33s.  Time for last 10,000,000:   93s.  Last read position: chr10:4,972,662\n",
      "INFO\t2023-02-21 16:16:47\tSortSam\tRead    20,000,000 records.  Elapsed time: 00:03:00s.  Time for last 10,000,000:   86s.  Last read position: chr11:129,859,095\n",
      "INFO\t2023-02-21 16:18:12\tSortSam\tRead    30,000,000 records.  Elapsed time: 00:04:25s.  Time for last 10,000,000:   85s.  Last read position: chr14:69,994,455\n",
      "INFO\t2023-02-21 16:19:31\tSortSam\tRead    40,000,000 records.  Elapsed time: 00:05:44s.  Time for last 10,000,000:   79s.  Last read position: chr17:28,642,022\n",
      "INFO\t2023-02-21 16:20:58\tSortSam\tRead    50,000,000 records.  Elapsed time: 00:07:11s.  Time for last 10,000,000:   87s.  Last read position: chr19:46,788,042\n",
      "INFO\t2023-02-21 16:22:25\tSortSam\tRead    60,000,000 records.  Elapsed time: 00:08:38s.  Time for last 10,000,000:   87s.  Last read position: chr21:39,349,258\n",
      "INFO\t2023-02-21 16:23:50\tSortSam\tRead    70,000,000 records.  Elapsed time: 00:10:03s.  Time for last 10,000,000:   85s.  Last read position: chr4:168,921,554\n",
      "INFO\t2023-02-21 16:25:16\tSortSam\tRead    80,000,000 records.  Elapsed time: 00:11:29s.  Time for last 10,000,000:   85s.  Last read position: chr6:116,277,125\n",
      "INFO\t2023-02-21 16:26:42\tSortSam\tRead    90,000,000 records.  Elapsed time: 00:12:55s.  Time for last 10,000,000:   86s.  Last read position: chr9:35,685,487\n",
      "INFO\t2023-02-21 16:27:51\tSortSam\tFinished reading inputs, merging and writing to output now.\n",
      "INFO\t2023-02-21 16:27:52\tSortingCollection\tCreating merging iterator from 196 files\n",
      "INFO\t2023-02-21 16:28:29\tSortSam\tWrote    10,000,000 records from a sorting collection.  Elapsed time: 00:14:42s.  Time for last 10,000,000:   36s.  Last read position: chr10:4,972,662\n",
      "INFO\t2023-02-21 16:29:05\tSortSam\tWrote    20,000,000 records from a sorting collection.  Elapsed time: 00:15:18s.  Time for last 10,000,000:   36s.  Last read position: chr11:129,859,095\n",
      "INFO\t2023-02-21 16:29:41\tSortSam\tWrote    30,000,000 records from a sorting collection.  Elapsed time: 00:15:54s.  Time for last 10,000,000:   36s.  Last read position: chr14:69,994,455\n",
      "INFO\t2023-02-21 16:30:18\tSortSam\tWrote    40,000,000 records from a sorting collection.  Elapsed time: 00:16:30s.  Time for last 10,000,000:   36s.  Last read position: chr17:28,642,022\n",
      "INFO\t2023-02-21 16:30:54\tSortSam\tWrote    50,000,000 records from a sorting collection.  Elapsed time: 00:17:07s.  Time for last 10,000,000:   36s.  Last read position: chr19:46,788,042\n",
      "INFO\t2023-02-21 16:31:30\tSortSam\tWrote    60,000,000 records from a sorting collection.  Elapsed time: 00:17:43s.  Time for last 10,000,000:   36s.  Last read position: chr21:39,349,258\n",
      "INFO\t2023-02-21 16:32:06\tSortSam\tWrote    70,000,000 records from a sorting collection.  Elapsed time: 00:18:19s.  Time for last 10,000,000:   36s.  Last read position: chr4:168,921,554\n",
      "INFO\t2023-02-21 16:32:43\tSortSam\tWrote    80,000,000 records from a sorting collection.  Elapsed time: 00:18:56s.  Time for last 10,000,000:   36s.  Last read position: chr6:116,277,125\n",
      "INFO\t2023-02-21 16:33:18\tSortSam\tWrote    90,000,000 records from a sorting collection.  Elapsed time: 00:19:31s.  Time for last 10,000,000:   35s.  Last read position: chr9:35,685,487\n",
      "[Tue Feb 21 16:33:47 CST 2023] picard.sam.SortSam done. Elapsed time: 20.01 minutes.\n",
      "Runtime.totalMemory()=851968000\n"
     ]
    }
   ],
   "source": [
    "!picard SortSam OUTPUT=new/data/alignments_clean_TOPHAT/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1_sort.bam INPUT=new/data/alignments_clean_TOPHAT/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1/accepted_hits.bam SORT_ORDER=queryname"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16:41:39.275 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/rsrch4/home/mol_cgenesis/nkdang/mambaforge/envs/rnaseq/share/picard-2.18.7-2/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
      "[Tue Feb 21 16:41:39 CST 2023] BuildBamIndex INPUT=new/data/alignments_clean_TOPHAT/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1_sort.bam    VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false GA4GH_CLIENT_SECRETS=client_secrets.json USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
      "[Tue Feb 21 16:41:39 CST 2023] Executing as nkdang@ldragon1 on Linux 3.10.0-1160.15.2.el7.x86_64 amd64; OpenJDK 64-Bit Server VM 1.8.0_332-b09; Deflater: Intel; Inflater: Intel; Provider GCS is not available; Picard version: 2.18.7-SNAPSHOT\n",
      "INFO\t2023-02-21 16:44:03\tBuildBamIndex\tSuccessfully wrote bam index file /rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/data/alignments_clean_TOPHAT/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1_sort.bai\n",
      "[Tue Feb 21 16:44:03 CST 2023] picard.sam.BuildBamIndex done. Elapsed time: 2.40 minutes.\n",
      "Runtime.totalMemory()=596115456\n"
     ]
    }
   ],
   "source": [
    "!picard BuildBamIndex INPUT=new/data/alignments_clean_TOPHAT/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1_sort.bam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A00330:132:HMNLKDSX5:2:2129:5791:28056\t163\tchrX\t1591729\t3\t41M1674N109M\t=\t1593480\t1901\tGCGGCGGCGGCGGCCTGGGACGCAGGCGGAGCCCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACACGTCTGAGGCCGTGGAGCTCTGCCCTGCTTACGGCTTGTACCTGAAGCCCATCACCAAGATG\tFFFFFFFFFFFFFFFFFF::FFFFFFFFFFFFFFF:FFFFFFFFFFFF:FFFFFFFFFFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFF\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:150\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591729\tHI:i:0\n",
      "A00330:132:HMNLKDSX5:2:1271:19940:28698\t393\tchrX\t1591741\t3\t29M1674N121M\t*\t0\t0\tGCCTGGGACGCAGGCGGAGCCCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACACGTCTGAGGCCGTGGAGCTCTGCCCTGCTTACGGCTTGTACCTGAAGCCCATCACCAAGATGACCATCAGCGTG\tFFFFFFFFFFFFFFFFF:FFFFF:FFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFF:FFFFFFFF\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:150\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591741\tHI:i:0\n",
      "A00330:132:HMNLKDSX5:2:2528:26910:18474\t163\tchrX\t1591745\t3\t25M1674N125M\t=\t1593608\t2012\tGGGACGCAGGCGGAGCCCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACACGTCTGAGGCCGTGGAGCTCTGCCCTGCTTACGGCTTGTACCTGAAGCCCATCACCAAGATGACCATCAGCGTGGCAC\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFF,FF:FFF:FFFFFFF:FFFFFFFFFFFFFFF:F:FFFFFFF,FFFF::FFF:,FFFFFFFF,FFFF,FFFFFFFFFFFFFFF:FFFFFFFFFFF\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:150\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591745\tHI:i:0\n",
      "A00330:132:HMNLKDSX5:2:1420:5593:31156\t163\tchrX\t1591754\t3\t16M1674N103M\t=\t1591754\t-1793\tGCGGAGCCCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACACGTCTGAGGCCGTGGAGCTCTGCCCTGCTTACGGCTTGTACCTGAAGCCCATCACC\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:119\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591754\tHI:i:0\n",
      "A00330:132:HMNLKDSX5:2:1420:5593:31156\t83\tchrX\t1591754\t3\t16M1674N103M\t=\t1591754\t-1793\tGCGGAGCCCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACACGTCTGAGGCCGTGGAGCTCTGCCCTGCTTACGGCTTGTACCTGAAGCCCATCACC\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFF\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:119\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591754\tHI:i:0\n",
      "A00330:132:HMNLKDSX5:2:1322:31629:11992\t419\tchrX\t1591762\t3\t8M1674N71M\t=\t1591762\t-1753\tCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACACGTCTGAGGCCGTGGAGCTCTGCC\tFFFFFFFFFFFFFFFFFFFFFFFFFF:F,FFFFFFFFFFFFFF:FFFFFFF:FF,F:F:FFFFFFFFFFFFFFFF:FFF\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:79\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591762\tHI:i:0\n",
      "A00330:132:HMNLKDSX5:2:2539:15881:2942\t419\tchrX\t1591762\t3\t8M1674N71M\t=\t1591762\t-1753\tCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACACGTCTGAGGCCGTGGAGCTCTGCC\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:79\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591762\tHI:i:0\n",
      "A00330:132:HMNLKDSX5:2:1322:31629:11992\t339\tchrX\t1591762\t3\t8M1674N71M\t=\t1591762\t-1753\tCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACACGTCTGAGGCCGTGGAGCTCTGCC\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:79\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591762\tHI:i:0\n",
      "A00330:132:HMNLKDSX5:2:2205:15609:24173\t89\tchrX\t1591762\t3\t8M1674N71M\t*\t0\t0\tCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACCCGTCTGAGGCCGTGGAGCTCTGCC\tFFFFF:FFFF:FF:FFFF,F,FFFF:FFF::FFFF:,:FF:,FF:FFFFF:FF,,FF:,FFFF,F:F:F:FF,F,FF:F\tAS:i:-1\tXN:i:0\tXM:i:1\tXO:i:0\tXG:i:0\tNM:i:1\tMD:Z:54A24\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591762\tHI:i:0\n",
      "A00330:132:HMNLKDSX5:2:2539:15881:2942\t339\tchrX\t1591762\t3\t8M1674N71M\t=\t1591762\t-1753\tCCGCGCAGGCCCAAGGTCCCGGAGGCTATGGCAGCGGCTACCATCGTGCACGACACGTCTGAGGCCGTGGAGCTCTGCC\tFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\tAS:i:0\tXN:i:0\tXM:i:0\tXO:i:0\tXG:i:0\tNM:i:0\tMD:Z:79\tYT:Z:UU\tXS:A:+\tNH:i:2\tCC:Z:chrY\tCP:i:1591762\tHI:i:0\n",
      "samtools view: writing to standard output failed: Broken pipe\n",
      "samtools view: error closing standard output: -1\n"
     ]
    }
   ],
   "source": [
    "!samtools view new/data/alignments_clean_TOPHAT/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1/Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1_sorted.bam \"chrX:1591604-1602520\" | head"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>21415-P5-OE-3</th>\n",
       "      <th>31284-P5-OE-2</th>\n",
       "      <th>31325-KI-2</th>\n",
       "      <th>31329-KI-1</th>\n",
       "      <th>31332-KI-3</th>\n",
       "      <th>31351-P5-OE-1</th>\n",
       "      <th>31371-KO-2</th>\n",
       "      <th>31372-KO-1</th>\n",
       "      <th>31373-KO-3</th>\n",
       "      <th>31379-WT-3</th>\n",
       "      <th>31385-WT-1</th>\n",
       "      <th>31387-WT-2</th>\n",
       "      <th>31391-P5-WT-1</th>\n",
       "      <th>31395-P5-WT-3</th>\n",
       "      <th>31396-P5-WT-2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Geneid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000102693.2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000064842.3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000051951.6</th>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000102851.2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000103377.2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000095523.2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000095475.2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000094855.2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.17</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000095019.2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSMUSG00000095041.8</th>\n",
       "      <td>1945.79</td>\n",
       "      <td>1587.33</td>\n",
       "      <td>2092.0</td>\n",
       "      <td>1976.5</td>\n",
       "      <td>1886.0</td>\n",
       "      <td>1914.25</td>\n",
       "      <td>1428.5</td>\n",
       "      <td>2047.5</td>\n",
       "      <td>2290.0</td>\n",
       "      <td>2479.0</td>\n",
       "      <td>1921.67</td>\n",
       "      <td>1619.5</td>\n",
       "      <td>1423.67</td>\n",
       "      <td>1569.5</td>\n",
       "      <td>1602.75</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>56980 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     21415-P5-OE-3 31284-P5-OE-2 31325-KI-2 31329-KI-1  \\\n",
       "Geneid                                                                   \n",
       "ENSMUSG00000102693.2           0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000064842.3           0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000051951.6           2.0           0.0        1.0        0.0   \n",
       "ENSMUSG00000102851.2           0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000103377.2           0.0           0.0        0.0        0.0   \n",
       "...                            ...           ...        ...        ...   \n",
       "ENSMUSG00000095523.2           0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000095475.2           0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000094855.2           0.0          0.17        0.0        0.0   \n",
       "ENSMUSG00000095019.2           0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000095041.8       1945.79       1587.33     2092.0     1976.5   \n",
       "\n",
       "                     31332-KI-3 31351-P5-OE-1 31371-KO-2 31372-KO-1  \\\n",
       "Geneid                                                                \n",
       "ENSMUSG00000102693.2        0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000064842.3        0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000051951.6        0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000102851.2        0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000103377.2        0.0           0.0        0.0        0.0   \n",
       "...                         ...           ...        ...        ...   \n",
       "ENSMUSG00000095523.2        0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000095475.2        0.0           0.0        0.0       0.67   \n",
       "ENSMUSG00000094855.2       0.08           0.0        0.0        0.0   \n",
       "ENSMUSG00000095019.2        0.0           0.0        0.0        0.0   \n",
       "ENSMUSG00000095041.8     1886.0       1914.25     1428.5     2047.5   \n",
       "\n",
       "                     31373-KO-3 31379-WT-3 31385-WT-1 31387-WT-2  \\\n",
       "Geneid                                                             \n",
       "ENSMUSG00000102693.2        0.0        0.0        0.0        0.0   \n",
       "ENSMUSG00000064842.3        0.0        0.0        0.0        0.0   \n",
       "ENSMUSG00000051951.6        2.0        0.0        6.0        0.0   \n",
       "ENSMUSG00000102851.2        0.0        0.0        0.0        0.0   \n",
       "ENSMUSG00000103377.2        0.0        0.0        0.0        0.0   \n",
       "...                         ...        ...        ...        ...   \n",
       "ENSMUSG00000095523.2        0.0        0.0        0.0        0.0   \n",
       "ENSMUSG00000095475.2        0.0        0.0        0.0        0.0   \n",
       "ENSMUSG00000094855.2        0.0        0.0       0.17        0.0   \n",
       "ENSMUSG00000095019.2        0.0        0.0        0.0        0.0   \n",
       "ENSMUSG00000095041.8     2290.0     2479.0    1921.67     1619.5   \n",
       "\n",
       "                     31391-P5-WT-1 31395-P5-WT-3 31396-P5-WT-2  \n",
       "Geneid                                                          \n",
       "ENSMUSG00000102693.2           0.0           0.0           0.0  \n",
       "ENSMUSG00000064842.3           0.0           0.0           0.0  \n",
       "ENSMUSG00000051951.6           0.0           2.0           2.0  \n",
       "ENSMUSG00000102851.2           0.0           0.0           0.0  \n",
       "ENSMUSG00000103377.2           0.0           0.0           0.0  \n",
       "...                            ...           ...           ...  \n",
       "ENSMUSG00000095523.2           0.0           0.0           0.0  \n",
       "ENSMUSG00000095475.2           0.0           0.0           0.0  \n",
       "ENSMUSG00000094855.2           0.0           0.0           0.0  \n",
       "ENSMUSG00000095019.2           0.0           0.0           0.0  \n",
       "ENSMUSG00000095041.8       1423.67        1569.5       1602.75  \n",
       "\n",
       "[56980 rows x 15 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mouse_count = pd.read_csv(\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/featureCounts/mouses.csv\", sep='\\t', comment=\"#\")\n",
    "mouse_count = mouse_count.drop(['Chr','Start','End','Strand','Length'], axis=1).T\n",
    "mouse_count.index = [\"Geneid\"]+[Path(i).parent.name for i in mouse_count.index if i!=\"Geneid\"]\n",
    "mouse_count = mouse_count.T\n",
    "mouse_count = mouse_count.set_index('Geneid')\n",
    "mouse_count = mouse_count.reindex(sorted(mouse_count.columns), axis=1)\n",
    "mouse_count.to_csv(\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/featureCounts/mouses_clean.csv\")\n",
    "mouse_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21415-P5-OE-3        59167261.47\n",
       "31284-P5-OE-2    57552619.259999\n",
       "31325-KI-2            63075531.3\n",
       "31329-KI-1           66386473.76\n",
       "31332-KI-3       59680117.479999\n",
       "31351-P5-OE-1        55772850.81\n",
       "31371-KO-2           63176251.93\n",
       "31372-KO-1           62003451.87\n",
       "31373-KO-3       60665152.559999\n",
       "31379-WT-3           63284193.43\n",
       "31385-WT-1           62849611.48\n",
       "31387-WT-2           60959321.83\n",
       "31391-P5-WT-1        60457772.08\n",
       "31395-P5-WT-3        61329998.16\n",
       "31396-P5-WT-2        49060591.14\n",
       "dtype: object"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mouse_count.sum(axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hela-WT_DMSO_4_day__Replicate_1</th>\n",
       "      <th>Hela-WT_DMSO_4_day__Replicate_2</th>\n",
       "      <th>Hela-WT_DMSO_4_day__Replicate_3</th>\n",
       "      <th>Hela-WT_EPZ_4_day__Replicate_1</th>\n",
       "      <th>Hela-WT_EPZ_4_day__Replicate_2</th>\n",
       "      <th>Hela-WT_EPZ_4_day__Replicate_3</th>\n",
       "      <th>Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1</th>\n",
       "      <th>Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_2</th>\n",
       "      <th>Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_3</th>\n",
       "      <th>Hela_sgSART3-KO__4__Pos_pCW-GFP-SART3-W377A_Replicate_1</th>\n",
       "      <th>...</th>\n",
       "      <th>RFH838-DMSOR-DMSO-R3</th>\n",
       "      <th>RFH838-DMSOR-MRTX-R1</th>\n",
       "      <th>RFH838-DMSOR-MRTX-R2</th>\n",
       "      <th>RFH838-DMSOR-MRTX-R3</th>\n",
       "      <th>RFH838-MRTXR-DMSO-R1</th>\n",
       "      <th>RFH838-MRTXR-DMSO-R2</th>\n",
       "      <th>RFH838-MRTXR-DMSO-R3</th>\n",
       "      <th>RFH838-MRXTR-MRTX-R1</th>\n",
       "      <th>RFH838-MRXTR-MRTX-R2</th>\n",
       "      <th>RFH838-MRXTR-MRTX-R3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Geneid</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000290825.1</th>\n",
       "      <td>1.67</td>\n",
       "      <td>1.33</td>\n",
       "      <td>1.25</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.37</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.67</td>\n",
       "      <td>...</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.83</td>\n",
       "      <td>13.67</td>\n",
       "      <td>36.08</td>\n",
       "      <td>26.17</td>\n",
       "      <td>16.39</td>\n",
       "      <td>10.62</td>\n",
       "      <td>43.09</td>\n",
       "      <td>15.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000223972.6</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000227232.5</th>\n",
       "      <td>577.07</td>\n",
       "      <td>854.79</td>\n",
       "      <td>749.95</td>\n",
       "      <td>1334.09</td>\n",
       "      <td>1043.06</td>\n",
       "      <td>980.55</td>\n",
       "      <td>834.86</td>\n",
       "      <td>899.36</td>\n",
       "      <td>304.08</td>\n",
       "      <td>419.49</td>\n",
       "      <td>...</td>\n",
       "      <td>842.46</td>\n",
       "      <td>704.85</td>\n",
       "      <td>1033.78</td>\n",
       "      <td>1451.4</td>\n",
       "      <td>803.61</td>\n",
       "      <td>1057.17</td>\n",
       "      <td>871.44</td>\n",
       "      <td>1232.96</td>\n",
       "      <td>920.44</td>\n",
       "      <td>1683.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000278267.1</th>\n",
       "      <td>7.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>12.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>9.5</td>\n",
       "      <td>18.0</td>\n",
       "      <td>26.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>31.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000243485.5</th>\n",
       "      <td>1.4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.95</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5.75</td>\n",
       "      <td>2.5</td>\n",
       "      <td>1.9</td>\n",
       "      <td>15.7</td>\n",
       "      <td>1.08</td>\n",
       "      <td>7.57</td>\n",
       "      <td>1.67</td>\n",
       "      <td>3.15</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000276017.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.67</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.33</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000278817.1</th>\n",
       "      <td>149.83</td>\n",
       "      <td>146.17</td>\n",
       "      <td>115.0</td>\n",
       "      <td>267.75</td>\n",
       "      <td>226.17</td>\n",
       "      <td>256.83</td>\n",
       "      <td>182.67</td>\n",
       "      <td>173.67</td>\n",
       "      <td>44.17</td>\n",
       "      <td>114.67</td>\n",
       "      <td>...</td>\n",
       "      <td>185.53</td>\n",
       "      <td>112.0</td>\n",
       "      <td>188.33</td>\n",
       "      <td>190.5</td>\n",
       "      <td>137.63</td>\n",
       "      <td>153.67</td>\n",
       "      <td>137.5</td>\n",
       "      <td>185.83</td>\n",
       "      <td>98.17</td>\n",
       "      <td>314.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000277196.4</th>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.5</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7.5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>9.5</td>\n",
       "      <td>7.0</td>\n",
       "      <td>14.0</td>\n",
       "      <td>16.0</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000278625.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000277374.1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>62750 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Hela-WT_DMSO_4_day__Replicate_1  \\\n",
       "Geneid                                              \n",
       "ENSG00000290825.1                            1.67   \n",
       "ENSG00000223972.6                             0.0   \n",
       "ENSG00000227232.5                          577.07   \n",
       "ENSG00000278267.1                             7.0   \n",
       "ENSG00000243485.5                             1.4   \n",
       "...                                           ...   \n",
       "ENSG00000276017.1                             0.0   \n",
       "ENSG00000278817.1                          149.83   \n",
       "ENSG00000277196.4                             4.0   \n",
       "ENSG00000278625.1                             0.0   \n",
       "ENSG00000277374.1                             0.0   \n",
       "\n",
       "                  Hela-WT_DMSO_4_day__Replicate_2  \\\n",
       "Geneid                                              \n",
       "ENSG00000290825.1                            1.33   \n",
       "ENSG00000223972.6                             0.0   \n",
       "ENSG00000227232.5                          854.79   \n",
       "ENSG00000278267.1                             4.0   \n",
       "ENSG00000243485.5                             0.0   \n",
       "...                                           ...   \n",
       "ENSG00000276017.1                             0.0   \n",
       "ENSG00000278817.1                          146.17   \n",
       "ENSG00000277196.4                             0.0   \n",
       "ENSG00000278625.1                             0.0   \n",
       "ENSG00000277374.1                             0.0   \n",
       "\n",
       "                  Hela-WT_DMSO_4_day__Replicate_3  \\\n",
       "Geneid                                              \n",
       "ENSG00000290825.1                            1.25   \n",
       "ENSG00000223972.6                             0.0   \n",
       "ENSG00000227232.5                          749.95   \n",
       "ENSG00000278267.1                             7.0   \n",
       "ENSG00000243485.5                             0.0   \n",
       "...                                           ...   \n",
       "ENSG00000276017.1                            0.67   \n",
       "ENSG00000278817.1                           115.0   \n",
       "ENSG00000277196.4                             0.0   \n",
       "ENSG00000278625.1                             0.0   \n",
       "ENSG00000277374.1                             0.0   \n",
       "\n",
       "                  Hela-WT_EPZ_4_day__Replicate_1  \\\n",
       "Geneid                                             \n",
       "ENSG00000290825.1                            0.0   \n",
       "ENSG00000223972.6                            0.0   \n",
       "ENSG00000227232.5                        1334.09   \n",
       "ENSG00000278267.1                           26.0   \n",
       "ENSG00000243485.5                           0.95   \n",
       "...                                          ...   \n",
       "ENSG00000276017.1                            0.0   \n",
       "ENSG00000278817.1                         267.75   \n",
       "ENSG00000277196.4                            8.0   \n",
       "ENSG00000278625.1                            0.0   \n",
       "ENSG00000277374.1                            0.0   \n",
       "\n",
       "                  Hela-WT_EPZ_4_day__Replicate_2  \\\n",
       "Geneid                                             \n",
       "ENSG00000290825.1                            0.0   \n",
       "ENSG00000223972.6                            0.0   \n",
       "ENSG00000227232.5                        1043.06   \n",
       "ENSG00000278267.1                           13.0   \n",
       "ENSG00000243485.5                            0.0   \n",
       "...                                          ...   \n",
       "ENSG00000276017.1                            0.0   \n",
       "ENSG00000278817.1                         226.17   \n",
       "ENSG00000277196.4                            7.0   \n",
       "ENSG00000278625.1                            0.0   \n",
       "ENSG00000277374.1                            0.0   \n",
       "\n",
       "                  Hela-WT_EPZ_4_day__Replicate_3  \\\n",
       "Geneid                                             \n",
       "ENSG00000290825.1                            1.5   \n",
       "ENSG00000223972.6                            0.0   \n",
       "ENSG00000227232.5                         980.55   \n",
       "ENSG00000278267.1                            8.0   \n",
       "ENSG00000243485.5                            0.0   \n",
       "...                                          ...   \n",
       "ENSG00000276017.1                           0.25   \n",
       "ENSG00000278817.1                         256.83   \n",
       "ENSG00000277196.4                            0.0   \n",
       "ENSG00000278625.1                            0.0   \n",
       "ENSG00000277374.1                            0.0   \n",
       "\n",
       "                  Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_1  \\\n",
       "Geneid                                                                 \n",
       "ENSG00000290825.1                                               1.37   \n",
       "ENSG00000223972.6                                                0.0   \n",
       "ENSG00000227232.5                                             834.86   \n",
       "ENSG00000278267.1                                                3.0   \n",
       "ENSG00000243485.5                                                1.0   \n",
       "...                                                              ...   \n",
       "ENSG00000276017.1                                                3.0   \n",
       "ENSG00000278817.1                                             182.67   \n",
       "ENSG00000277196.4                                                3.0   \n",
       "ENSG00000278625.1                                                0.0   \n",
       "ENSG00000277374.1                                                0.0   \n",
       "\n",
       "                  Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_2  \\\n",
       "Geneid                                                                 \n",
       "ENSG00000290825.1                                               0.33   \n",
       "ENSG00000223972.6                                                0.0   \n",
       "ENSG00000227232.5                                             899.36   \n",
       "ENSG00000278267.1                                               12.0   \n",
       "ENSG00000243485.5                                                0.0   \n",
       "...                                                              ...   \n",
       "ENSG00000276017.1                                               1.33   \n",
       "ENSG00000278817.1                                             173.67   \n",
       "ENSG00000277196.4                                                1.5   \n",
       "ENSG00000278625.1                                                0.0   \n",
       "ENSG00000277374.1                                                0.0   \n",
       "\n",
       "                  Hela_sgSART3-KO__4__Pos_pCW-GFP-EV_M2__Replicate_3  \\\n",
       "Geneid                                                                 \n",
       "ENSG00000290825.1                                                0.3   \n",
       "ENSG00000223972.6                                                0.0   \n",
       "ENSG00000227232.5                                             304.08   \n",
       "ENSG00000278267.1                                                0.0   \n",
       "ENSG00000243485.5                                                0.0   \n",
       "...                                                              ...   \n",
       "ENSG00000276017.1                                                0.0   \n",
       "ENSG00000278817.1                                              44.17   \n",
       "ENSG00000277196.4                                                5.0   \n",
       "ENSG00000278625.1                                                0.0   \n",
       "ENSG00000277374.1                                                0.0   \n",
       "\n",
       "                  Hela_sgSART3-KO__4__Pos_pCW-GFP-SART3-W377A_Replicate_1  \\\n",
       "Geneid                                                                      \n",
       "ENSG00000290825.1                                               0.67        \n",
       "ENSG00000223972.6                                                0.0        \n",
       "ENSG00000227232.5                                             419.49        \n",
       "ENSG00000278267.1                                                0.0        \n",
       "ENSG00000243485.5                                                0.0        \n",
       "...                                                              ...        \n",
       "ENSG00000276017.1                                                0.0        \n",
       "ENSG00000278817.1                                             114.67        \n",
       "ENSG00000277196.4                                                2.0        \n",
       "ENSG00000278625.1                                                0.0        \n",
       "ENSG00000277374.1                                                0.0        \n",
       "\n",
       "                   ... RFH838-DMSOR-DMSO-R3 RFH838-DMSOR-MRTX-R1  \\\n",
       "Geneid             ...                                             \n",
       "ENSG00000290825.1  ...                 11.0                  1.0   \n",
       "ENSG00000223972.6  ...                  0.0                  0.0   \n",
       "ENSG00000227232.5  ...               842.46               704.85   \n",
       "ENSG00000278267.1  ...                  9.5                 18.0   \n",
       "ENSG00000243485.5  ...                 5.75                  2.5   \n",
       "...                ...                  ...                  ...   \n",
       "ENSG00000276017.1  ...                  0.0                  0.0   \n",
       "ENSG00000278817.1  ...               185.53                112.0   \n",
       "ENSG00000277196.4  ...                  7.5                  2.0   \n",
       "ENSG00000278625.1  ...                  0.0                  0.0   \n",
       "ENSG00000277374.1  ...                  0.0                  0.0   \n",
       "\n",
       "                  RFH838-DMSOR-MRTX-R2 RFH838-DMSOR-MRTX-R3  \\\n",
       "Geneid                                                        \n",
       "ENSG00000290825.1                 8.83                13.67   \n",
       "ENSG00000223972.6                  0.0                  0.0   \n",
       "ENSG00000227232.5              1033.78               1451.4   \n",
       "ENSG00000278267.1                 26.0                  5.0   \n",
       "ENSG00000243485.5                  1.9                 15.7   \n",
       "...                                ...                  ...   \n",
       "ENSG00000276017.1                 0.25                  0.0   \n",
       "ENSG00000278817.1               188.33                190.5   \n",
       "ENSG00000277196.4                  0.0                 13.0   \n",
       "ENSG00000278625.1                  0.0                  0.0   \n",
       "ENSG00000277374.1                  0.0                  0.0   \n",
       "\n",
       "                  RFH838-MRTXR-DMSO-R1 RFH838-MRTXR-DMSO-R2  \\\n",
       "Geneid                                                        \n",
       "ENSG00000290825.1                36.08                26.17   \n",
       "ENSG00000223972.6                  0.0                 14.0   \n",
       "ENSG00000227232.5               803.61              1057.17   \n",
       "ENSG00000278267.1                  6.0                 13.0   \n",
       "ENSG00000243485.5                 1.08                 7.57   \n",
       "...                                ...                  ...   \n",
       "ENSG00000276017.1                  0.0                  0.0   \n",
       "ENSG00000278817.1               137.63               153.67   \n",
       "ENSG00000277196.4                  3.0                  9.5   \n",
       "ENSG00000278625.1                  0.0                  0.0   \n",
       "ENSG00000277374.1                  0.0                  0.0   \n",
       "\n",
       "                  RFH838-MRTXR-DMSO-R3 RFH838-MRXTR-MRTX-R1  \\\n",
       "Geneid                                                        \n",
       "ENSG00000290825.1                16.39                10.62   \n",
       "ENSG00000223972.6                  0.0                  0.0   \n",
       "ENSG00000227232.5               871.44              1232.96   \n",
       "ENSG00000278267.1                  7.0                 20.0   \n",
       "ENSG00000243485.5                 1.67                 3.15   \n",
       "...                                ...                  ...   \n",
       "ENSG00000276017.1                  0.0                  0.0   \n",
       "ENSG00000278817.1                137.5               185.83   \n",
       "ENSG00000277196.4                  7.0                 14.0   \n",
       "ENSG00000278625.1                  0.0                  0.0   \n",
       "ENSG00000277374.1                  0.0                  0.0   \n",
       "\n",
       "                  RFH838-MRXTR-MRTX-R2 RFH838-MRXTR-MRTX-R3  \n",
       "Geneid                                                       \n",
       "ENSG00000290825.1                43.09                 15.0  \n",
       "ENSG00000223972.6                  0.0                  6.0  \n",
       "ENSG00000227232.5               920.44              1683.28  \n",
       "ENSG00000278267.1                 20.0                 31.0  \n",
       "ENSG00000243485.5                  0.0                  0.0  \n",
       "...                                ...                  ...  \n",
       "ENSG00000276017.1                  0.0                  0.0  \n",
       "ENSG00000278817.1                98.17                314.5  \n",
       "ENSG00000277196.4                 16.0                 21.0  \n",
       "ENSG00000278625.1                  0.0                  0.0  \n",
       "ENSG00000277374.1                  0.0                  0.0  \n",
       "\n",
       "[62750 rows x 42 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_count = pd.read_csv(\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/featureCounts/humans.csv\", sep='\\t', comment=\"#\")\n",
    "human_count = human_count.drop(['Chr','Start','End','Strand','Length'], axis=1).T\n",
    "human_count.index = [\"Geneid\"]+[Path(i).parent.name for i in human_count.index if i!=\"Geneid\"]\n",
    "human_count = human_count.T\n",
    "human_count = human_count.set_index('Geneid')\n",
    "human_count = human_count.reindex(sorted(human_count.columns), axis=1)\n",
    "human_count.to_csv(\"/rsrch4/home/mol_cgenesis/EMC_BIC_rsrch4/nkdang/Splicing_Projects/new/featureCounts/humans_clean.csv\")\n",
    "human_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "rnaseq",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "f466e0a83e05f58a6811402e4bba84a917e14410c8ae6f873f70d9ea619f8ec5"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
